src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -215,6 +217,25 @@ namespace xrt::core::hip
     free(host_src);
   }
 
+
+  static void
+  hip_memcpy_host2device_async(hipDeviceptr_t dst, void* src, size_t size, hipStream_t stream)  
+  {
+    if (!src)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid src pointer in hipMemCpyH2DAsync");
+
+    auto hip_mem_dst = memory_database::instance().get_hip_mem_from_addr(dst);
+    if (!hip_mem_dst)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid destination handle in hipMemCpyH2DAsync");
+
+    auto hip_stream = get_stream(stream);
+    if (!hip_stream)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid stream handle in hipMemCpyH2DAsync");
+   
+    copy_buffer cp_buf(hip_stream, XCL_BO_SYNC_BO_TO_DEVICE, hip_mem_dst, src);
```

Comment:
You should need to create a shared pointer of command and push it to the stream . You should not submit the command in this function directly.
Please check hip_module_launch_kernel function for reference.

Reply:
fixed.